### DISCUSSION TITLE
How to properly enable depth test on Android with Lit model?

### DISCUSSION THREAD

**Discussion by ndming** [google/filament#6598] [Q&A] [answered]
My rendered 3D shape on Android appears to have depth test issue. I notice that the `depthCulling` responsible for the depth test is enabled by default in the material guideline. Do I have to set up anything else to have Filament do the depth test properly? Here's my material code:

```
material {
    name : lit,

    shadingModel : lit,

    blending : fade,

    parameters : [
        {
            type : float3,
            name : rgb
        },
        {
            type : float,
            name : alpha
        },
        {
            type : float,
            name : metallic
        },
        {
            type : float,
            name : roughness
        },
        {
            type : float,
            name : reflectance
        },
        {
            type : float,
            name : clearCoat
        },
        {
            type : float,
            name : clearCoatRoughness
        },
        {
            type : float,
            name : anisotropy
        }
    ],
}

fragment {
    void material(inout MaterialInputs material) {
        prepareMaterial(material);

        material.baseColor.rgb = materialParams.rgb;
        material.baseColor.a = materialParams.alpha;
        material.metallic = materialParams.metallic;
        material.roughness = materialParams.roughness;
        material.reflectance = materialParams.reflectance;
        material.clearCoat = materialParams.clearCoat;
        material.clearCoatRoughness = materialParams.clearCoatRoughness;
        material.anisotropy = materialParams.anisotropy;

        material.baseColor.rgb *= materialParams.alpha;
    }
}
```

**Comment by romainguy** [comment, marked as answer]
What issue do you have exactly? Depth testing is on by default. However you are using transparency and that can cause issues since triangles are not sorted individually. You can look at transparencyMode in the materials documentation to help with that.

**Reply by ndming** [reply]
Here are the captures of the issue. The geometry appears normal in certain viewing directions and buggy in others. Is it because I'm modeling concave shapes? I tried convex geometries and they are fine.

![issue](https://user-images.githubusercontent.com/61007865/221780969-23f7206f-0a15-4bbd-b24d-b3730c186d12.png)

![normal](https://user-images.githubusercontent.com/61007865/221781718-71e802d9-7cb4-45b8-82d5-4f59ecc17140.png)

**Reply by ndming** [reply]
After I applied `transparency : twoPassesOneSide` to the material code as you pointed out, things resolved nicely. Thanks for helping.
